src/workerd/jsg/modules.h

```diff
@@ -171,9 +171,9 @@ class ModuleRegistry {
     ModuleInfo& operator=(ModuleInfo&&) = default;
   };
 
-  virtual kj::Maybe<ModuleInfo&> resolve(const kj::Path& specifier) = 0;
+  virtual kj::Maybe<ModuleInfo&> resolve(v8::Isolate* isolate, const kj::Path& specifier) = 0;
```

Comment:
Let's have `resolve` take a `jsg::Lock& js` instead of the isolate

Reply:
Unfortunately that would mean changing very long call chains, since they don't have lock wired through.

E.g.:

resolve <- evaluateSyntheticModuleCallback <- createSyntheticModule <- ModuleInfo with no lock in sight yet.

Let me know if you want to create locks before calling resolve or tackle passing jsg::Lock everywhere in modules.h

Reply:
I'll make separate PR with this cleanup today if you don't mind.